### PR TITLE
Adding scraping support for GoodReads platform (GSSoC'23)

### PR DESCRIPTION
**Pull request by brohithkr**
## Description

This pr adds `BookScraper` class that helps in scraping details of book based on book title and author name or book id from [Goodreads](https://goodreads.com) Platform.  

- `Boookscraper` class has been added in books.py in goodreads folder
- documentation has been updated to reflect corresponding changes
- appropriate error handing mechanisms have been implemented

## Checklist

- [x] Have you adhered to the repository's defined coding convention rules?
- [x] Have you updated the 'documentation.md' file with the method/function documentation?
- [x] Have you sent a message along with the result or response?
- [x] Have you used the try-catch technique?
- [x] Has the method/class been added to the documentation (md file)?

## Screenshots

image

## Additional Notes/Comments

None

---

I certify that I have carried out the relevant checks and provided the requisite screenshot for validation by submitting this pull request.
I appreciate your contribution.

### CODE REVIEW

File: src/scrape_up/goodreads/books.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
+from selenium import webdriver
+from selenium.webdriver.chrome.service import Service
+from webdriver_manager.chrome import ChromeDriverManager
```
Comment (nikhil25803) [inline src/scrape_up/goodreads/books.py:3]: Hey @brohithkr | `ChromeDriverManager` does not work with the packages on all the devices, it requires installation of the chrome driver. 
Many of our modules are facing the same issue, try to look for an alternative.

**Review by nikhil25803 — CHANGES_REQUESTED** [review]

File: src/scrape_up/goodreads/books.py (hunk 1 above)
Comment (nikhil25803) [reply src/scrape_up/goodreads/books.py:3]: Give it a try

Comment (nikhil25803) [reply src/scrape_up/goodreads/books.py:3]: The problem is, apart from download it needs to be set to the path in system's environment variables.

File: src/scrape_up/goodreads/books.py [hunk 2]
```diff
@@ -69,7 +69,12 @@ def get_soup(self):
...
         """
         BASE_DIR = os.path.dirname(os.path.abspath(__file__))
-        driver_path = os.path.join(BASE_DIR, r"\Drivers\.wdm\drivers\chromedriver\win32\114.0.5735.90\chromedriver.exe")
+        driver_path = ""
+        if "/" in BASE_DIR:
+            driver_path = os.path.join(BASE_DIR, r"/Drivers/.wdm/drivers/chromedriver/win32/114.0.5735.90/chromedriver.exe")
+        else:
+            driver_path = os.path.join(BASE_DIR, r"\Drivers\.wdm\drivers\chromedriver\win32\114.0.5735.90\chromedriver.exe")
+
         driver = webdriver.Chrome(
```
Comment (brohithkr) [inline src/scrape_up/goodreads/books.py:78]: @nikhil25803 I added absolute path of chrome driver so we don't need to change environment variables.

**Comment by brohithkr** [comment]
@nikhil25803 I added absolute path of driver to webdriver.Chrome() so we dont have to change environment variables

**Review by nikhil25803 — COMMENTED** [review]
Can't accept the PR, as these many additions will make the page heavier also it won't work because it needs to be installed on `pip`.

The only solution is try finding an alternative to `ChromeDriver`, try the soup generated by `BS4` only.

**Comment by brohithkr** [comment]
@nikhil25803 But all the modules I imported are already in system requirements. So users will have already installed it using `pip install -r requirements.txt`. Also Goodreads data is behind javascript. So we need selenium to execute the javascript. And also I am just opening the page using selenium and taking the page source after javascript is executed. So I am mostly using bs4. Only place I used selenium to get html after js is executed.

**Comment by brohithkr** [comment]
hey @nikhil25803 we could do one thing. We could add a script for installling chrome driver and then we can use the chromedriver by the path method. Since selenium is being used by many scripts we could do this so that even other scripts could utilize the chromedriver installed by user.

**Comment by nikhil25803** [comment]
This wont work on cloud services.

**Pull request closed without merging**